### ISSUE TITLE
az acr build still fails on decoding UTF characters

### ISSUE THREAD

**Issue by favoretti** [Azure/azure-cli#20766]
## Describe the bug

**Command Name**
`az acr build`

**Errors:**
```
ERROR: 'ascii' codec can't encode character '\u2714' in position 125: ordinal not in range(128)
Traceback (most recent call last):
  File "/opt/az/lib/python3.6/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 658, in execute
    raise ex
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 721, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 692, in _run_job
    result = cmd_copy(params)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/acr/build.py", line 143, in acr_build
    return stream_logs(cmd, client, run_id, registry_name, resource_group_name, timeout, no_format, True)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/acr/_stream_utils.py", line 67, in stream_logs
    raise_error_on_failure)
  File "/opt/az/lib/python3.6/site-packages/azure/cli/command_modules/acr/_stream_utils.py", line 143, in _stream_logs
    print(flush.decode('utf-8', errors='ignore'))
UnicodeEncodeError: 'ascii' codec can't encode character '\u2714' in position 125: ordinal not in range(128)
To open an issue, please run: 'az feedback'


```

It fails on the line with the first checkbox:
![image](https://user-images.githubusercontent.com/805046/146532314-3061546f-f8a7-4359-8e61-27487cfd3411.png)

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az acr build --registry {} --agent-pool {} --no-format -t {} -f {} {} --subscription {}`

## Environment Summary
Azure DevOps ACR build agents

```
Starting: Build docker container with ACR
==============================================================================
Task         : Azure CLI
Description  : Run Azure CLI commands against an Azure subscription in a PowerShell Core/Shell script when running on Linux agent or PowerShell/PowerShell Core/Batch script when running on Windows agent.
Version      : 2.1.2
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/deploy/azure-cli
==============================================================================
/usr/bin/az --version

azure-cli                         2.31.0
Please let us know how we are doing: https://aka.ms/azureclihats
and let us know if you're interested in trying out our newest features: https://aka.ms/CLIUXstudy

core                              2.31.0
telemetry                          1.0.6

Python location '/opt/az/bin/python3'
Extensions directory '/root/.azure/cliextensions'

Python (Linux) 3.6.10 (default, Dec  3 2021, 05:08:37) 
[GCC 7.5.0]

Legal docs and information: aka.ms/AzureCliLegal


Your CLI is up-to-date.
```

**Comment by favoretti** [comment]
Before anyone suggests to disable UTF output in Angular CLI - it's not that simple and yes, we've tried :) Current workaround is to run build with `--no-logs`, but it's a very sub-optimal situation, we want to see build logs of course.

**Comment by favoretti** [comment]
Ok, found a workaround that seems to fix it though.

```
- task: AzureCLI@2
      displayName: Build docker container with ACR
      inputs:
        azureSubscription: <sub>
        scriptType: bash
        scriptLocation: inlineScript
        inlineScript: >-
          apt-get update &&
          apt-get install locales &&
          locale-gen en_US.UTF-8 &&
          export LANG="en_US.UTF-8" &&
          export LANGUAGE="en_US:en" &&
          export LC_ALL="en_US.UTF-8" &&
          az acr build
          --registry {}
          --agent-pool {}
          -t {}
          -f {}
          {}
```

**Comment by jiasli** [comment]
Routing to service team to help take a look. Please expect delay due to the holiday season.

**Comment by northtyphoon** [comment]
@favoretti Python has a long-standing issue with POSIX locale. I think the workaround you mentioned did the trick

The following article suggests reconfigure the system locale.

https://chriswarrick.com/blog/2017/06/18/unix-locales-vs-unicode/

The following article suggests that Python 3.7 has done some improvement for POSIX locale.

https://vstinner.github.io/python37-new-utf8-mode.html

**Comment by navba-MSFT** [comment]
@favoretti Apologies for the late reply. I hope you had a chance to look at the above comment by @northtyphoon .
Please let us know if you need any further assistance on this issue. Awaiting your reply.

**Comment by navba-MSFT** [comment]
@favoretti I wanted to do quick follow-up to check if you had a chance to look at my above comment. This github issue is open for quite sometime. Please let us know if you need any further assistance on this. Awaiting your reply.

**Comment by favoretti** [comment]
Oh, apologies, missed notifications on this one. Yeah, since i got a workaround, I guess this can be closed.

**Comment by ckarrasalithya** [comment]
@favoretti I think the bug should be reopen, as the workaround is only applicable in a specific situation (Angular CLI), and we have encountered the same bug in several other situations, when running Python and R commands. Each situation needs a new workaround and we have to clutter our code with these workarounds. 

For example:
* pip needs to be used with --progress-bar off
* We also had to change the method used to install pip, because the script provided for installation displayed progress bars we were unable to turn off
* Piping stderr and stdout to iconv to transliterate non-ASCII characters
* ...
And we might encounter other situations where another workaround is needed as we keep developing our containers.

Thanks
